### DISCUSSION TITLE
getModelForClass and model method not typed?

### DISCUSSION THREAD

**Discussion by anlauren** [typegoose/typegoose#463] [Q&A] [answered]
I must be doing something wrong, but i'm just using the base code from the documentation. when doing the following code, everything is returning "any" type:

```typescript
import {prop, getModelForClass} from "@typegoose/typegoose";

class KittenClass {
  @prop()
  public name?: string;
}

async function Test() {
  const Kitten = getModelForClass(KittenClass);

  let document = await Kitten.create({ name: "Kitty" });
}

```
<img width="493" alt="Screenshot 2020-12-28 at 12 00 09" src="https://user-images.githubusercontent.com/7627816/103209811-48805980-4904-11eb-9f18-ae7088d49fe5.png">
<img width="540" alt="Screenshot 2020-12-28 at 12 00 17" src="https://user-images.githubusercontent.com/7627816/103209825-4e763a80-4904-11eb-9797-5e5d9e93a408.png">

## Versions

- System: macos 
- NodeJS: 12.20.0
- Typescript: 4.1.3

- Compiler / Transpiler: ts-node 
- Typegoose(NPM): ^7.4.5 
- mongoose: ^5.11.8 
- mongodb: 3.6.3

---

**Comment by hasezoey** [comment, marked as answer]
invalid mongoose version, highest supported is `mongoose@5.10.18`, and typegoose requires `@types/mongoose`

**Comment by anlauren** [comment]
All right works well with that change thank you!

Maybe it would be good to change the docs to specify `mongoose <=5.10.18`, because it says `mongoose ^5.10.4`, which i think means any version in the range `5.x.x` :)

**Comment by hasezoey** [comment]
> Maybe it would be good to change the docs to specify mongoose <=5.10.18, because it says mongoose ^5.10.4, which i think means any version in the range 5.x.x :)

thats true, #432
